### PR TITLE
pep8 and docstring

### PR DESCRIPTION
**Pull request by kontogiannisg**

### CODE REVIEW

File: crossai/ai/nn1d/_bilstm_time.py [hunk 1]
```diff
@@ -1,54 +1,57 @@
...
+    n_layers: int = 3,
+    lstm_units: list = [32, 32, 32],
+    dense_units: list = [128],
+    drp_rate: float = 0.,
+    spatial: bool = False,
+    mc_inference: Union[bool, None] = None
+) -> tf.keras.Model:
     """Constructs a deep neural network using bidirectional LSTM.
 
     This model converts low-level audio features into high-level expressions.
```
Comment (tzamalisp) [inline crossai/ai/nn1d/_bilstm_time.py]: @gkontogiannhs only audio? I think we have to say "time series"?

File: crossai/ai/nn1d/_bilstm_time.py [hunk 2]
```diff
@@ -1,54 +1,57 @@
...
+    num_classes: int = 1,
+    classifier_activation: Union[str, Callable] = "softmax",
+    n_layers: int = 3,
+    lstm_units: list = [32, 32, 32],
+    dense_units: list = [128],
+    drp_rate: float = 0.,
+    spatial: bool = False,
+    mc_inference: Union[bool, None] = None
+) -> tf.keras.Model:
     """Constructs a deep neural network using bidirectional LSTM.
```
Comment (tzamalisp) [inline crossai/ai/nn1d/_bilstm_time.py:22]: @gkontogiannhs -> LSTM (BiLSTM) layers

File: crossai/ai/_layers_dropout.py [hunk 3]
```diff
@@ -1,16 +1,40 @@
...
 from tensorflow.keras.layers import Dropout, SpatialDropout1D, SpatialDropout2D
-from tensorflow.keras.layers import Dense
+from tensorflow.keras.layers import Layer, Dense
+from tensorflow import keras
 
 
 @keras.saving.register_keras_serializable(package="MyLayers")
 class MCDropout(Dropout):
+    """
+    Monte Carlo Dropout layer.
```
Comment (tzamalisp) [inline crossai/ai/_layers_dropout.py]: @gkontogiannhs add how it works at a high level (also in the next functions). Also, describe why we add these layers, except for using them for model generalization (e.g., prediction -> measuring robustness)

File: crossai/ai/nn1d/_xceptiontime.py [hunk 4]
```diff
@@ -39,45 +45,36 @@ def XceptionTime(input_shape,
...
+        kernel_regularize: L2 regularization factor; can be a string or float.
+            If a string is provided, it's converted to a float.
+        kernel_constraint: Max norm constraint for the kernel weights.
+        spatial: If true, applies Spatial Dropout, else applies standard
+            Dropout.
+        mc_inference: If True, enables Monte Carlo dropout during inference.
+            Defaults to None.
 
     Returns:
-        A Keras Model instance.
```
Comment (tzamalisp) [inline crossai/ai/nn1d/_xceptiontime.py:74]: @gkontogiannhs  a Keras or TensorFlow?

File: crossai/ai/nn1d/_xceptiontime.py [hunk 5]
```diff
@@ -165,45 +162,52 @@ def XceptionTime(input_shape,
...
+    act_func: str = "relu",
+    drp_on: bool = False,
+    drp_rate: float = 0.5,
+    spatial: bool = True,
+    mc_inference: bool = None,
+    kernel_initialize: str = None,
+    kernel_regularize: float = None,
+    kernel_constraint: int = None
+):
     """Create a block of layers consisting of Conv1D, BatchNormalization,
```
Comment (tzamalisp) [inline crossai/ai/nn1d/_xceptiontime.py]: @gkontogiannhs Creates...

File: crossai/ai/nn2d/_resnet34.py [hunk 6]
```diff
@@ -1,63 +1,62 @@
...
+    dense_layers: int = 0,
+    dense_units: List[int] = [128, 128],
+    dropout: bool = False,
+    dropout_first: bool = False,
+    dropout_rate: List[float] = [0.5, 0.5],
+    spatial: bool = False,
+    mc_inference: Union[bool, None] = None
+) -> tf.keras.Model:
+    """
+    Constructs a ResNet34 model suitable for image classification tasks.
```
Comment (tzamalisp) [inline crossai/ai/nn2d/_resnet34.py]: @gkontogiannhs This should be in the line above -> """Constructs..

File: crossai/ai/nn1d/_bilstm_time.py [hunk 7]
```diff
@@ -90,17 +98,16 @@ def bilstm_block(inputs, n_layers, lstm_units, drp_rate, mc_inference=None):
     return x
 
 
-def dense_block(inputs, n_layers, dense_units):
-    """
-    Builds a block of dense layers.
+def dense_block(inputs: tf.Tensor, n_layers: int, dense_units: list):
```
Comment (tzamalisp) [inline crossai/ai/nn1d/_bilstm_time.py]: @gkontogiannhs split in different lines

File: crossai/ai/nn1d/_xceptiontime.py [hunk 8]
```diff
@@ -225,25 +229,29 @@ def conv1d_block(inputs, nf, ks=1, strd=1, pad="same", bias=False, bn=True,
...
+def xception_block(
+    inputs: tf.Tensor,
+    n_filters: int,
+    depth: int = 4,
+    use_residual: bool = True,
+    kernel_initialize: bool = None,
+    kernel_regularize: bool = None,
+    kernel_constraint: bool = None
+):
     """Xception Block function that applies a series of Xception modules,
```
Comment (tzamalisp) [inline crossai/ai/nn1d/_xceptiontime.py]: @gkontogiannhs Xception Block function that --> remove that one. We say "Applies a..."

File: crossai/ai/nn1d/_xceptiontime.py [hunk 9]
```diff
@@ -274,36 +282,38 @@ def xception_block(inputs, n_filters, depth=4, use_residual=True,
...
+    inputs: tf.Tensor,
+    n_filters: int,
+    use_bottleneck: bool = True,
+    kernel_size: int = 41,
+    stride: int = 1,
+    kernel_initialize: str = None,
+    kernel_regularize: float = None,
+    kernel_constraint: int = None
+):
     """XceptionModule function that applies a series of SeparableConv1D layers
```
Comment (tzamalisp) [inline crossai/ai/nn1d/_xceptiontime.py]: @gkontogiannhs "Applies the Xception module which is a series...."

**Comment by tzamalisp** [comment]
@gkontogiannhs I suggest the following changes in all the AI modeling parts, in general:
* In some arguments, the description starts in lowercase while it should be in uppercase
* We should say, "An integer that represents the number of filters..." instead of "Number of filters...."
* When we describe the function, the description should start after the `"""` characters and not in the next line. Please, be careful with the indentation. There is no consistency in the way of writing.
* Check where the parentheses indentation when closing def's arguments
* We proposed in the meeting that the docstring will include the papers' links.


Check this example

```
def fetch_smalltable_rows(
    table_handle: smalltable.Table,
    keys: Sequence[bytes | str],
    require_all_keys: bool = False,
) -> Mapping[bytes, tuple[str, ...]]:
    """Fetches rows from a Smalltable.

    Retrieves rows pertaining to the given keys from the Table instance
    represented by table_handle.  String keys will be UTF-8 encoded.

    Args:
        table_handle: An open smalltable.Table instance.
        keys: A sequence of strings representing the key of each table
          row to fetch.  String keys will be UTF-8 encoded.
        require_all_keys: If True only rows with values set for all keys will be
          returned.

    Returns:
        A dict mapping keys to the corresponding table row data
        fetched. Each row is represented as a tuple of strings. For
        example:

        {b'Serak': ('Rigel VII', 'Preparer'),
         b'Zim': ('Irk', 'Invader'),
         b'Lrrr': ('Omicron Persei 8', 'Emperor')}

        Returned keys are always bytes.  If a key from the keys argument is
        missing from the dictionary, then that row was not found in the
        table (and require_all_keys must have been False).

    Raises:
        IOError: An error occurred accessing the smalltable.
    """

```

**Review by tzamalisp — CHANGES_REQUESTED** [review]

File: crossai/ai/nn1d/_bilstm_time.py (hunk 2 above)
Comment (kontogiannisg) [reply crossai/ai/nn1d/_bilstm_time.py:22]: Returned value is a keras model.

**Pull request merged**